frontend/src/hooks/useLocationByIdOrNull.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from "react";
+import { Location } from "../api";
+import { useLocationFinderApi } from "../services/apiClients/ApiClientsContext";
+
+const locationCache = new Map<string, Location>();
+
+export function useLocationByIdOrNull(locationId: string | null): Location | null {
+  const locationFinderApi = useLocationFinderApi();
+  const [location, setLocation] = useState<Location | null>(null);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    checkNull(locationId)
+      || checkCache(locationId)
+      || fetchLocation(locationId);
```

Comment:
I would not write that in this way. This causes side effects that are not really visible here. My recommendation: Move all setLocation calls out of their functions and use earily exits